### PR TITLE
fix logic error about instanced buffer

### PR DESCRIPTION
**Pull request by minggo**

### CODE REVIEW

File: cocos/renderer/pipeline/InstancedBuffer.cpp [hunk 1]
```diff
@@ -58,6 +58,23 @@ void InstancedBuffer::destroyInstancedBuffer() {
     InstancedBuffer::buffers.clear();
 }
 
+void InstancedBuffer::destoryInstancedBuffer(scene::Pass *pass) {
```
Comment (dumganhar) [inline cocos/renderer/pipeline/InstancedBuffer.cpp]: Typo: destory -> destroy

File: cocos/renderer/pipeline/InstancedBuffer.cpp [hunk 2]
```diff
@@ -58,6 +58,23 @@ void InstancedBuffer::destroyInstancedBuffer() {
...
+    }
+    
+    auto &bufferMap = iter->second;
+    auto iter2 = bufferMap.find(0);
+    if (iter2 == bufferMap.end()) {
+        return;
+    }
+    
+    InstancedBuffer *instancedBuffer = iter2->second;
+    delete instancedBuffer;
```
Comment (dumganhar) [inline cocos/renderer/pipeline/InstancedBuffer.cpp]: delete -> CC_DELETE

**Review by dumganhar — APPROVED** [review]

File: cocos/renderer/pipeline/InstancedBuffer.cpp (hunk 2 above)
Comment (minggo) [reply cocos/renderer/pipeline/InstancedBuffer.cpp]: 改了

File: cocos/renderer/pipeline/InstancedBuffer.cpp (hunk 1 above)
Comment (minggo) [reply cocos/renderer/pipeline/InstancedBuffer.cpp]: 好眼力

**Pull request merged**